gui/src/ScenarioSelector.tsx

```diff
@@ -106,14 +125,47 @@ const ScenarioNodes = ({ scenarios = [], showPrimary = true }: ScenarioNodesProp
           nodeId={id}
           label={
             showPrimary && primary ? (
-              <Badge badgeContent={<FlagOutlined sx={FlagSx} />} color="primary" anchorOrigin={BadgePos} sx={BadgeSx}>
-                <Grid container alignItems="center" direction="row" flexWrap="nowrap" justifyContent="flex-start" spacing={1}>
+              <Badge
+                badgeContent={<FlagOutlined sx={FlagSx} />}
+                color="primary"
+                anchorOrigin={BadgePos}
+                sx={BadgeSx}
+              >
+                <Grid
```

Comment:
This could be made as another internal React component for readability (and to prevent repetition)